compiler/resolution/functionResolution.cpp

```diff
@@ -6595,6 +6595,7 @@ static Expr* resolveExprPhase2(Expr* origExpr, FnSymbol* fn, Expr* expr) {
 
       } else {
         resolveFunction(call->resolvedFunction());
+        resolveAlsoParallelIterators(call->resolvedFunction(), call);
```

Comment:
I'd instead make CallExpr* forCall be an optional argument (that defaults to NULL) to resolveFunction.
Then, I'd call resolveAlsoParallelIterators from resolveFunction, just after protoIteratorClass is called:

```
        insertAndResolveCasts(fn);

        if (fn->isIterator() == true && fn->iteratorInfo == NULL) {
          protoIteratorClass(fn, yieldedType);

        } else if (fn->hasFlag(FLAG_TYPE_CONSTRUCTOR) == true) {
          resolveTypeConstructor(fn);

        }

        // Resolve also leader and standalone iterators
        if (fn->isIterator()) {
          INT_ASSERT(forCall != NULL);
          INT_ASSERT(fn->iteratorInfo != NULL);
          resolveAlsoParallelIterators(fn, forCall);
        }
```

Reply:
Even though you are not storing the IteratorGroup in the IteratorInfo, I think the code flow will be more reasonable if you can move resolveAlsoParallelIterators into resolveFunction. If that can't work for some reason, OK.